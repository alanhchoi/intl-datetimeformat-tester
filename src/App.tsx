import { format, parse } from 'date-fns';
import { useMemo } from 'react';
import { useForm } from 'react-hook-form';
import Modal from 'react-modal';
import tw, { styled } from 'twin.macro';

import Result from './Result';
import { DateTimeFormatOptions } from './types';
import filterEmptyProperties from './filterEmptyProperties';
import Select from './components/Select';
import Heading from './components/Heading';
import Input from './components/Input';
import Link from './components/Link';
import Button from './components/Button';

type FormFieldValues = {
  mode: 'default' | 'range';
  date: string;
  time: string;
  endDate: string;
  endTime: string;
} & DateTimeFormatOptions;

type DateTimeFormatOption<T extends keyof Intl.DateTimeFormatOptions> =
  NonNullable<Intl.DateTimeFormatOptions[T]>;

type HourCycle = DateTimeFormatOption<'hourCycle'>;
type DateStyle = DateTimeFormatOption<'dateStyle'>;
type TimeStyle = DateTimeFormatOption<'timeStyle'>;
type DayPeriod = DateTimeFormatOption<'dayPeriod'>;
type Weekday = DateTimeFormatOption<'weekday'>;
type Year = DateTimeFormatOption<'year'>;
type Month = DateTimeFormatOption<'month'>;
type Day = DateTimeFormatOption<'day'>;
type Hour = DateTimeFormatOption<'hour'>;
type Minute = DateTimeFormatOption<'minute'>;
type Second = DateTimeFormatOption<'second'>;
type TimeZoneName = DateTimeFormatOption<'timeZoneName'>;

const hourCycles: HourCycle[] = ['h11', 'h12', 'h23', 'h24'];
const dateStyles: DateStyle[] = ['full', 'long', 'medium', 'short'];
const timeStyles: TimeStyle[] = ['full', 'long', 'medium', 'short'];
const dayPeriods: DayPeriod[] = ['long', 'short', 'narrow'];
const weekdays: Weekday[] = ['long', 'short', 'narrow'];
const years: Year[] = ['numeric', '2-digit'];
const months: Month[] = ['numeric', '2-digit', 'long', 'short', 'narrow'];
const days: Day[] = ['numeric', '2-digit'];
const hours: Hour[] = ['numeric', '2-digit'];
const minutes: Minute[] = ['numeric', '2-digit'];
const seconds: Second[] = ['numeric', '2-digit'];
const timeZoneNames: TimeZoneName[] = ['long', 'short'];

enum Boolean {
  True = 'true',
  False = 'false',
}

const rootElement = document.getElementById('root');

if (rootElement) {
  Modal.setAppElement(rootElement);
}

const Container = tw.div`p-12`;

const Layout = styled.div`
  ${tw`gap-x-8 flex flex-row`}

  @media (max-width: 80rem) {
    ${tw`flex-col`}
    grid-template-columns: 12rem minmax(640px, 1fr);
    grid-template-areas: 'form' 'aside results';
  }
`;

const Form = styled.form`
  ${tw`flex flex-col space-y-10`}
  min-width: 640px;
  flex: 1;
  grid-area: form;
`;

const Label = tw.label`font-mono w-40 text-right px-6`;

const SectionHeading = styled.h2`
  ${tw`mb-1 text-xl font-bold`}
`;

const FormRow = styled.div`
  ${tw`flex items-baseline`}

  > *:not(label) {
    flex: 1;
  }
`;

const DateTimeRow = styled.div`
  ${tw`flex items-baseline space-x-0.5`}
`;

const DateSection = styled.section`
  ${tw`flex flex-wrap`}
`;

const RadioButton = styled.input`
  display: none;

  & + label {
    ${tw`flex items-center justify-center box-border rounded px-4 h-11 text-base font-medium text-gray-800 select-none`}
  }

  &:not(:checked) + label {
    ${tw`hover:underline cursor-pointer`}
  }

  &:checked + label {
    ${tw`font-bold border-2 border-gray-800`}
  }
`;

const FormatOptions = styled.section`
  ${tw`flex flex-wrap`}

  grid-column: 1 / span 2;
  grid-row: 2 / span 1;

  > div {
    ${tw`w-1/2 space-y-3 min-w-0`}
  }
`;

const Aside = styled.aside`
  grid-area: aside;
`;

const Results = styled.section`
  flex: 1;
  min-width: 0;
`;

const DateTimeInput = tw(Input)`w-44`;

const OptionSelect = tw(Select)`font-mono font-bold`;

const App = () => {
  const { register, watch, getValues, reset } = useForm<FormFieldValues>({
    defaultValues: {
      mode: 'default',
      date: format(new Date(), 'yyyy-MM-dd'),
      time: format(new Date(), 'HH:mm:ss'),
      endDate: format(Date.now() + 86400000, 'yyyy-MM-dd'),
      endTime: format(Date.now() + 86400000, 'HH:mm:ss'),
    },
  });

  const {
    mode,
    date,
    time,
    endDate,
    endTime,
    hourCycle,
    dateStyle,
    timeStyle,
    dayPeriod,
    weekday,
    year,
    month,
    day,
    hour,
    minute,
    second,
    timeZoneName,
    timeZone,
    hour12,
  } = watch();

  const formatOptions: DateTimeFormatOptions = useMemo(
    () => ({
      hourCycle,
      dateStyle,
      timeStyle,
      dayPeriod,
      weekday,
      year,
      month,
      day,
      hour,
      minute,
      second,
      timeZoneName,
      timeZone,
      hour12,
    }),
    [
      dateStyle,
      day,
      dayPeriod,
      hour,
      hour12,
      hourCycle,
      minute,
      month,
      second,
      timeStyle,
      timeZone,
      timeZoneName,
      weekday,
      year,
    ]
  );

  const renderSelect = ({
    label,
    id: idOrUndefined,
    name,
    options,
    isIdIsName = true,
    includeEmptyOption = true,
  }: {
    label: string;
    id?: string;
    name: keyof FormFieldValues;
    options: string[];
    isIdIsName?: boolean;
    includeEmptyOption?: boolean;
  }) => {
    const id = isIdIsName ? name : idOrUndefined;

    return (
      <FormRow>
        <Label htmlFor={id}>{label}</Label>
        <OptionSelect variant="solid" {...register(name)} id={id}>
          {includeEmptyOption && (
            <option key="" value="">
              -
            </option>
          )}
          {options.map((item) => (
            <option key={item} value={item}>
              {item}
            </option>
          ))}
        </OptionSelect>
      </FormRow>
    );
  };

  const options = filterEmptyProperties(formatOptions);

  const clearAllSelections = () => {
    const { mode, date, time, endDate, endTime } = getValues();
    reset({ mode, date, time, endDate, endTime });
  };

  return (
    <Container>
      <Heading tw="mb-8 whitespace-nowrap">Intl.DateTimeFormat Tester</Heading>

      <Layout>
        <Form>
          <DateSection>
            <SectionHeading tw="w-full mb-3">Date</SectionHeading>
            <fieldset tw="w-40 flex">
              <p>
                <RadioButton
                  id="mode__default"
                  type="radio"
                  {...register('mode')}
                  value="default"
                />
                <label htmlFor="mode__default">Time</label>
              </p>
              <p>
                <RadioButton
                  id="mode__range"
                  type="radio"
                  {...register('mode')}
                  value="range"
                />
                <label htmlFor="mode__range">Range</label>
              </p>
            </fieldset>

            <div tw="space-y-3 flex-1">
              <DateTimeRow>
                <Label htmlFor="date" tw="w-auto flex-1">
                  {mode === 'default' ? 'Date' : 'Start'}
                </Label>
                <DateTimeInput
                  variant="solid"
                  id="date"
                  {...register('date')}
                  type="date"
                  aria-label="Date"
                />
                <DateTimeInput
                  variant="solid"
                  id="time"
                  {...register('time')}
                  type="time"
                  step="1"
                  aria-label="Time"
                />
              </DateTimeRow>

              {mode === 'range' && (
                <DateTimeRow>
                  <Label htmlFor="endDate" tw="w-auto flex-1">
                    End
                  </Label>
                  <DateTimeInput
                    variant="solid"
                    id="endDate"
                    {...register('endDate')}
                    type="date"
                  />
                  <DateTimeInput
                    variant="solid"
                    id="time"
                    {...register('endTime')}
                    type="time"
                    step="1"
                    aria-label="Time"
                  />
                </DateTimeRow>
              )}
            </div>
          </DateSection>

          <FormatOptions>
            <header tw="flex justify-between w-full mb-2">
              <SectionHeading tw="w-full mb-3 flex flex-col">
                Format Options
                <div tw="text-base font-normal">
                  Learn more about these options at{' '}
                  <Link href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#parameters">
                    MDN Web Docs
                  </Link>
                  .
                </div>
              </SectionHeading>
              <Button
                type="button"
                variant="outline"
                size="sm"
                onClick={clearAllSelections}
              >
                Clear all selections
              </Button>
            </header>
            {renderSelect({
              label: 'dateStyle',
              name: 'dateStyle',
              options: dateStyles,
            })}

            {renderSelect({
              label: 'timeStyle',
              name: 'timeStyle',
              options: timeStyles,
            })}

            {renderSelect({
              label: 'timeZoneName',
              name: 'timeZoneName',
              options: timeZoneNames,
            })}

            <FormRow>
              <Label htmlFor="timeZone">timeZone</Label>
              <Input
                variant="solid"
                id="timeZone"
                type="text"
                {...register('timeZone')}
              />
            </FormRow>

            {renderSelect({
              label: 'dayPeriod',
              name: 'dayPeriod',
              options: dayPeriods,
            })}

            {renderSelect({
              label: 'hour12',
              name: 'hour12',
              options: [Boolean.True, Boolean.False],
            })}

            {renderSelect({
              label: 'hourCycle',
              name: 'hourCycle',
              options: hourCycles,
            })}

            {renderSelect({
              label: 'weekday',
              name: 'weekday',
              options: weekdays,
            })}

            {renderSelect({
              label: 'year',
              name: 'year',
              options: years,
            })}

            {renderSelect({
              label: 'month',
              name: 'month',
              options: months,
            })}

            {renderSelect({
              label: 'day',
              name: 'day',
              options: days,
            })}

            {renderSelect({
              label: 'hour',
              name: 'hour',
              options: hours,
            })}

            {renderSelect({
              label: 'minute',
              name: 'minute',
              options: minutes,
            })}

            {renderSelect({
              label: 'second',
              name: 'second',
              options: seconds,
            })}
          </FormatOptions>
        </Form>

        <Results>
          <SectionHeading tw="mb-3">Results</SectionHeading>
          <Result
            mode={mode}
            options={options}
            date={parse(`${date} ${time}`, 'yyyy-MM-dd HH:mm:ss', new Date())}
            endDate={parse(
              `${endDate} ${endTime}`,
              'yyyy-MM-dd HH:mm:ss',
              new Date()
            )}
          />
        </Results>
      </Layout>
    </Container>
  );
};

export default App;
