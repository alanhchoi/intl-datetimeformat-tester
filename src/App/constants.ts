type DateTimeFormatOption<T extends keyof Intl.DateTimeFormatOptions> =
  NonNullable<Intl.DateTimeFormatOptions[T]>;

type HourCycle = DateTimeFormatOption<'hourCycle'>;
type DateStyle = DateTimeFormatOption<'dateStyle'>;
type TimeStyle = DateTimeFormatOption<'timeStyle'>;
type DayPeriod = DateTimeFormatOption<'dayPeriod'>;
type Weekday = DateTimeFormatOption<'weekday'>;
type Year = DateTimeFormatOption<'year'>;
type Month = DateTimeFormatOption<'month'>;
type Day = DateTimeFormatOption<'day'>;
type Hour = DateTimeFormatOption<'hour'>;
type Minute = DateTimeFormatOption<'minute'>;
type Second = DateTimeFormatOption<'second'>;
type TimeZoneName = DateTimeFormatOption<'timeZoneName'>;

export enum Hour12Options {
  True = 'true',
  False = 'false',
}

export const hourCycles: HourCycle[] = ['h11', 'h12', 'h23', 'h24'];
export const dateStyles: DateStyle[] = ['full', 'long', 'medium', 'short'];
export const timeStyles: TimeStyle[] = ['full', 'long', 'medium', 'short'];
export const dayPeriods: DayPeriod[] = ['long', 'short', 'narrow'];
export const weekdays: Weekday[] = ['long', 'short', 'narrow'];
export const years: Year[] = ['numeric', '2-digit'];
export const months: Month[] = [
  'numeric',
  '2-digit',
  'long',
  'short',
  'narrow',
];
export const days: Day[] = ['numeric', '2-digit'];
export const hours: Hour[] = ['numeric', '2-digit'];
export const minutes: Minute[] = ['numeric', '2-digit'];
export const seconds: Second[] = ['numeric', '2-digit'];
export const timeZoneNames: TimeZoneName[] = ['long', 'short'];
