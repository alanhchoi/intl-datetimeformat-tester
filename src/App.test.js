import { render } from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  jest.useFakeTimers('modern');
  jest.setSystemTime(1620258815791);
  render(<App />);
  expect(document.body).toMatchSnapshot();
});
