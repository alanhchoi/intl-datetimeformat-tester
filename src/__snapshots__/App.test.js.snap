// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<body>
  <div>
    <div
      class="App__Container-sc-1grf95z-0 hnPEHr"
    >
      <h1
        class="sc-gKAaRy App___StyledHeading-sc-1grf95z-18 kNiOkG cElYlC"
      >
        Intl.DateTimeFormat Tester
      </h1>
      <div
        class="App__Layout-sc-1grf95z-1 kRTgqz"
      >
        <aside
          class="App__Aside-sc-1grf95z-15 ezuGlq"
        >
          <div>
            <h2
              class="Presets___StyledH-sc-12q1l4f-0 giZQDP"
              id="presets__title"
            >
              Presets
            </h2>
            <select
              aria-labelledby="presets__title"
              class="sc-jSFjdj Presets___StyledSelect-sc-12q1l4f-1 fVdCod jBjxLw"
              id="preset__select"
            >
              <option
                disabled=""
                selected=""
                value=""
              >
                No presets
              </option>
            </select>
            <hr
              class="sc-eCApnc kvfmdr"
            />
            <div
              class="Presets___StyledDiv-sc-12q1l4f-4 UpPZU"
            >
              <button
                class="sc-gtsrHT hncqzl"
                type="button"
              >
                Add to presets
              </button>
              <button
                class="sc-gtsrHT hncqzl"
                type="button"
              >
                Manage presets
              </button>
            </div>
          </div>
        </aside>
        <form
          class="App__Form-sc-1grf95z-2 iWXDnI"
        >
          <section
            class="App__DateSection-sc-1grf95z-12 boyKit"
          >
            <h2
              class="App__SectionHeading-sc-1grf95z-6 App___StyledSectionHeading-sc-1grf95z-9 iIAwIq eeXbHc"
            >
              Date
            </h2>
            <fieldset
              class="App___StyledFieldset-sc-1grf95z-19 bdvzaE"
            >
              <p>
                <input
                  class="App__RadioButton-sc-1grf95z-13 cOcsfv"
                  id="mode__default"
                  name="mode"
                  type="radio"
                  value="default"
                />
                <label
                  for="mode__default"
                >
                  Time
                </label>
              </p>
              <p>
                <input
                  class="App__RadioButton-sc-1grf95z-13 cOcsfv"
                  id="mode__range"
                  name="mode"
                  type="radio"
                  value="range"
                />
                <label
                  for="mode__range"
                >
                  Range
                </label>
              </p>
            </fieldset>
            <div
              class="App___StyledDiv-sc-1grf95z-20 gLTSDy"
            >
              <div
                class="App__DateTimeRow-sc-1grf95z-11 kCwicr"
              >
                <label
                  class="App__Label-sc-1grf95z-3 App___StyledLabel-sc-1grf95z-5 jVerDd ynYTQ"
                  for="date"
                >
                  Date
                </label>
                <input
                  aria-label="Date"
                  class="sc-hKFxyN HmHZy"
                  id="date"
                  name="date"
                  type="date"
                />
                <input
                  aria-label="Time"
                  class="sc-hKFxyN HmHZy"
                  id="time"
                  name="time"
                  step="1"
                  type="time"
                />
              </div>
            </div>
          </section>
          <section
            class="App__FormatOptions-sc-1grf95z-14 iwLfrM"
          >
            <h2
              class="App__SectionHeading-sc-1grf95z-6 App___StyledSectionHeading2-sc-1grf95z-8 iIAwIq jBqypm"
            >
              Format Options
              <span
                class="App___StyledSpan-sc-1grf95z-21 exKiFq"
              >
                Learn more about these options at
                 
                <a
                  class="sc-iCoGMd gQzRLD"
                  href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#parameters"
                >
                  MDN Web Docs
                </a>
                .
              </span>
            </h2>
            <div>
              <div
                class="App__FormRow-sc-1grf95z-10 ggpiSO"
              >
                <label
                  class="App__Label-sc-1grf95z-3 jVerDd"
                  for="dateStyle"
                >
                  dateStyle
                </label>
                <select
                  class="sc-jSFjdj App__OptionSelect-sc-1grf95z-17 emgxYk hyMvf"
                  id="dateStyle"
                  name="dateStyle"
                >
                  <option
                    value=""
                  >
                    -
                  </option>
                  <option
                    value="full"
                  >
                    full
                  </option>
                  <option
                    value="long"
                  >
                    long
                  </option>
                  <option
                    value="medium"
                  >
                    medium
                  </option>
                  <option
                    value="short"
                  >
                    short
                  </option>
                </select>
              </div>
              <div
                class="App__FormRow-sc-1grf95z-10 ggpiSO"
              >
                <label
                  class="App__Label-sc-1grf95z-3 jVerDd"
                  for="timeStyle"
                >
                  timeStyle
                </label>
                <select
                  class="sc-jSFjdj App__OptionSelect-sc-1grf95z-17 emgxYk hyMvf"
                  id="timeStyle"
                  name="timeStyle"
                >
                  <option
                    value=""
                  >
                    -
                  </option>
                  <option
                    value="full"
                  >
                    full
                  </option>
                  <option
                    value="long"
                  >
                    long
                  </option>
                  <option
                    value="medium"
                  >
                    medium
                  </option>
                  <option
                    value="short"
                  >
                    short
                  </option>
                </select>
              </div>
              <div
                class="App__FormRow-sc-1grf95z-10 ggpiSO"
              >
                <label
                  class="App__Label-sc-1grf95z-3 jVerDd"
                  for="timeZoneName"
                >
                  timeZoneName
                </label>
                <select
                  class="sc-jSFjdj App__OptionSelect-sc-1grf95z-17 emgxYk hyMvf"
                  id="timeZoneName"
                  name="timeZoneName"
                >
                  <option
                    value=""
                  >
                    -
                  </option>
                  <option
                    value="long"
                  >
                    long
                  </option>
                  <option
                    value="short"
                  >
                    short
                  </option>
                </select>
              </div>
              <div
                class="App__FormRow-sc-1grf95z-10 ggpiSO"
              >
                <label
                  class="App__Label-sc-1grf95z-3 jVerDd"
                  for="timeZone"
                >
                  timeZone
                </label>
                <input
                  class="sc-hKFxyN HmHZy"
                  id="timeZone"
                  name="timeZone"
                  type="text"
                />
              </div>
            </div>
            <div>
              <div
                class="App__FormRow-sc-1grf95z-10 ggpiSO"
              >
                <label
                  class="App__Label-sc-1grf95z-3 jVerDd"
                  for="dayPeriod"
                >
                  dayPeriod
                </label>
                <select
                  class="sc-jSFjdj App__OptionSelect-sc-1grf95z-17 emgxYk hyMvf"
                  id="dayPeriod"
                  name="dayPeriod"
                >
                  <option
                    value=""
                  >
                    -
                  </option>
                  <option
                    value="long"
                  >
                    long
                  </option>
                  <option
                    value="short"
                  >
                    short
                  </option>
                  <option
                    value="narrow"
                  >
                    narrow
                  </option>
                </select>
              </div>
              <div
                class="App__FormRow-sc-1grf95z-10 ggpiSO"
              >
                <label
                  class="App__Label-sc-1grf95z-3 jVerDd"
                  for="hour12"
                >
                  hour12
                </label>
                <select
                  class="sc-jSFjdj App__OptionSelect-sc-1grf95z-17 emgxYk hyMvf"
                  id="hour12"
                  name="hour12"
                >
                  <option
                    value=""
                  >
                    -
                  </option>
                  <option
                    value="true"
                  >
                    true
                  </option>
                  <option
                    value="false"
                  >
                    false
                  </option>
                </select>
              </div>
              <div
                class="App__FormRow-sc-1grf95z-10 ggpiSO"
              >
                <label
                  class="App__Label-sc-1grf95z-3 jVerDd"
                  for="hourCycle"
                >
                  hourCycle
                </label>
                <select
                  class="sc-jSFjdj App__OptionSelect-sc-1grf95z-17 emgxYk hyMvf"
                  id="hourCycle"
                  name="hourCycle"
                >
                  <option
                    value=""
                  >
                    -
                  </option>
                  <option
                    value="h11"
                  >
                    h11
                  </option>
                  <option
                    value="h12"
                  >
                    h12
                  </option>
                  <option
                    value="h23"
                  >
                    h23
                  </option>
                  <option
                    value="h24"
                  >
                    h24
                  </option>
                </select>
              </div>
              <div
                class="App__FormRow-sc-1grf95z-10 ggpiSO"
              >
                <label
                  class="App__Label-sc-1grf95z-3 jVerDd"
                  for="weekday"
                >
                  weekday
                </label>
                <select
                  class="sc-jSFjdj App__OptionSelect-sc-1grf95z-17 emgxYk hyMvf"
                  id="weekday"
                  name="weekday"
                >
                  <option
                    value=""
                  >
                    -
                  </option>
                  <option
                    value="long"
                  >
                    long
                  </option>
                  <option
                    value="short"
                  >
                    short
                  </option>
                  <option
                    value="narrow"
                  >
                    narrow
                  </option>
                </select>
              </div>
              <div
                class="App__FormRow-sc-1grf95z-10 ggpiSO"
              >
                <label
                  class="App__Label-sc-1grf95z-3 jVerDd"
                  for="year"
                >
                  year
                </label>
                <select
                  class="sc-jSFjdj App__OptionSelect-sc-1grf95z-17 emgxYk hyMvf"
                  id="year"
                  name="year"
                >
                  <option
                    value=""
                  >
                    -
                  </option>
                  <option
                    value="numeric"
                  >
                    numeric
                  </option>
                  <option
                    value="2-digit"
                  >
                    2-digit
                  </option>
                </select>
              </div>
              <div
                class="App__FormRow-sc-1grf95z-10 ggpiSO"
              >
                <label
                  class="App__Label-sc-1grf95z-3 jVerDd"
                  for="month"
                >
                  month
                </label>
                <select
                  class="sc-jSFjdj App__OptionSelect-sc-1grf95z-17 emgxYk hyMvf"
                  id="month"
                  name="month"
                >
                  <option
                    value=""
                  >
                    -
                  </option>
                  <option
                    value="numeric"
                  >
                    numeric
                  </option>
                  <option
                    value="2-digit"
                  >
                    2-digit
                  </option>
                  <option
                    value="long"
                  >
                    long
                  </option>
                  <option
                    value="short"
                  >
                    short
                  </option>
                  <option
                    value="narrow"
                  >
                    narrow
                  </option>
                </select>
              </div>
              <div
                class="App__FormRow-sc-1grf95z-10 ggpiSO"
              >
                <label
                  class="App__Label-sc-1grf95z-3 jVerDd"
                  for="day"
                >
                  day
                </label>
                <select
                  class="sc-jSFjdj App__OptionSelect-sc-1grf95z-17 emgxYk hyMvf"
                  id="day"
                  name="day"
                >
                  <option
                    value=""
                  >
                    -
                  </option>
                  <option
                    value="numeric"
                  >
                    numeric
                  </option>
                  <option
                    value="2-digit"
                  >
                    2-digit
                  </option>
                </select>
              </div>
              <div
                class="App__FormRow-sc-1grf95z-10 ggpiSO"
              >
                <label
                  class="App__Label-sc-1grf95z-3 jVerDd"
                  for="hour"
                >
                  hour
                </label>
                <select
                  class="sc-jSFjdj App__OptionSelect-sc-1grf95z-17 emgxYk hyMvf"
                  id="hour"
                  name="hour"
                >
                  <option
                    value=""
                  >
                    -
                  </option>
                  <option
                    value="numeric"
                  >
                    numeric
                  </option>
                  <option
                    value="2-digit"
                  >
                    2-digit
                  </option>
                </select>
              </div>
              <div
                class="App__FormRow-sc-1grf95z-10 ggpiSO"
              >
                <label
                  class="App__Label-sc-1grf95z-3 jVerDd"
                  for="minute"
                >
                  minute
                </label>
                <select
                  class="sc-jSFjdj App__OptionSelect-sc-1grf95z-17 emgxYk hyMvf"
                  id="minute"
                  name="minute"
                >
                  <option
                    value=""
                  >
                    -
                  </option>
                  <option
                    value="numeric"
                  >
                    numeric
                  </option>
                  <option
                    value="2-digit"
                  >
                    2-digit
                  </option>
                </select>
              </div>
              <div
                class="App__FormRow-sc-1grf95z-10 ggpiSO"
              >
                <label
                  class="App__Label-sc-1grf95z-3 jVerDd"
                  for="second"
                >
                  second
                </label>
                <select
                  class="sc-jSFjdj App__OptionSelect-sc-1grf95z-17 emgxYk hyMvf"
                  id="second"
                  name="second"
                >
                  <option
                    value=""
                  >
                    -
                  </option>
                  <option
                    value="numeric"
                  >
                    numeric
                  </option>
                  <option
                    value="2-digit"
                  >
                    2-digit
                  </option>
                </select>
              </div>
            </div>
          </section>
        </form>
        <section
          class="App__Results-sc-1grf95z-16 glkNVJ"
        >
          <h2
            class="App__SectionHeading-sc-1grf95z-6 App___StyledSectionHeading3-sc-1grf95z-7 iIAwIq zoEeP"
          >
            Results
          </h2>
          <div
            class="Result___StyledDiv-sc-1o5ucy3-0 jUKBLF"
          >
            <dd
              class="sc-bdnxRM Result___StyledCard-sc-1o5ucy3-1 gNaqGq jdYOMV"
            >
              <h3
                class="Result___StyledH-sc-1o5ucy3-2 jDEuIj"
              >
                en-US
              </h3>
              <div>
                5/6/2021
              </div>
            </dd>
            <dd
              class="sc-bdnxRM Result___StyledCard-sc-1o5ucy3-1 gNaqGq jdYOMV"
            >
              <h3
                class="Result___StyledH-sc-1o5ucy3-2 jDEuIj"
              >
                ko-KR
              </h3>
              <div
                class="PresetPreview___StyledDiv-u0tkt7-0 hTAIXz"
                role="alert"
              >
                <p
                  class="PresetPreview___StyledP-u0tkt7-1 eMZBjj"
                >
                  Something went wrong:
                </p>
                <div
                  class="PresetPreview___StyledDiv2-u0tkt7-2 StTlU"
                >
                  [@formatjs/intl Error MISSING_DATA] Missing locale data for locale: "ko-KR" in Intl.NumberFormat. Using default locale: "en" as fallback. See https://formatjs.io/docs/react-intl#runtime-requirements for more details 
                </div>
              </div>
            </dd>
          </div>
        </section>
      </div>
    </div>
  </div>
</body>
`;
